### ISSUE TITLE
Allow client setting overrides for host / host-patterns

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#996]
It would be nice if settings could be customized per host. To achieve that a section in the configuration is created with per-host overrides that should be chosen over the global the defaults. When the client pool is used these overrides should be applied automatically.

Suggestion how it could look like:

```
akka.http.host-connection-pool.per-host-override {
  "akka.io" = { # can use same things as in global `host-connection-pool` section
    max-connections = 10
  }

 "*.example.com" = { # allow `*` to apply overrides for all subdomains
 }
}
```

Related to https://github.com/akka/akka-http/issues/995.

**Comment by anilgursel** [comment]
One may also want to set different `akka.http.client.idle-timeout` (or other configuration) for different hosts.  How should the configuration be specified in that case?  I believe it would look like below:

```
akka.http { 
  host-connection-pool {
    max-connections = 20 # default for all other hosts
    per-host-override {
      "akka.io" = { # can use same things as in global `host-connection-pool` section
        max-connections = 10
      }
    }
  }

  client {
    idle-timeout = 20 s # default for all other hosts
    per-host-override {
      "akka.io" = { # can use same things as in global `host-connection-pool` section
        idle-timeout = 10 s
      }
    }
  }
} 
```

An alternative would be to specify host at the top level as below.  I believe it makes it easier to see what all overrides are there for a specific host:

```
akka.http {
  host-connection-pool.max-connections = 20
  client.idle-timeout = 20 s
} 

per-host-override."akka.io" = {
  akka.http {
    host-connection-pool.max-connections = 10
    client.idle-timeout = 10 s
  } 
}
```

**Comment by jrudolph** [comment]
Closed in #2574.
